ngmt/modules/td/_pham.py

```diff
@@ -0,0 +1,372 @@
+        for i in range(len(self.turns_all)):
+            # Check if the absolute value of the turn angle at index i is greater than or equal to 10
+            if abs(self.turns_all[i]) >= 10:
+                # Loop to search for potential hesitations
+                for j in range(i + 1, len(self.turns_all)):
+                    # Check if the difference between current index and i exceeds 4, or if the time between zero crossings exceeds half a second
+                    if (j - i) > 4 or (self.index_zero_crossings[j] - self.index_zero_crossings[i + 1] > (sampling_freq_Hz / 2)):
+                        # Break the loop if the conditions for hesitation are not met
+                        break
+                    else:
+                        # Check conditions for hesitation:
+                        # - Absolute values of both turns are greater than or equal to 10
+                        # - The relative change in yaw angle is less than 20% of the minimum turn angle
+                        # - The signs of both turns are the same
+                        if (abs(self.turns_all[i]) >= 10 and abs(self.turns_all[j]) >= 10 and
+                            abs(self.yaw[self.index_zero_crossings[i + 1]] - self.yaw[self.index_zero_crossings[j]]) /
+                            min(abs(self.turns_all[i]), abs(self.turns_all[j])) < 0.2 and
+                            np.sign(self.turns_all[i]) == np.sign(self.turns_all[j])):
+                            
+                            # Mark the range between i and j (inclusive) as a hesitation
+                            hesitation_markers[i:j+1] = 1
+                            # Break the inner loop as the hesitation condition is met
+                            break
+
+        # Initialize variables to store data related to turns without hesitation
+        sum_temp = 0  # Temporary sum for accumulating turn angles
+        turns_no_hesitation = []  # List to store turn angles without hesitation
+        flags_start_no_hesitation = []  # List to store start indices of turns without hesitation
+        flags_end_no_hesitation = []  # List to store end indices of turns without hesitation
+        durations_no_hesitation = []  # List to store durations of turns without hesitation
+        z = 1  # Index for keeping track of the current turn
+
+        # Iterate through each index in the hesitation_markers array
+        for i in range(len(hesitation_markers)):
+            # Check if there is a hesitation marker at the current index
+            if hesitation_markers[i] == 1:
+                # Check if sum_temp is zero, indicating the start of a new turn
+                if sum_temp == 0:
+                    f1 = self.index_zero_crossings[i]  # Store the start index of the turn
+                
+                # Check if the absolute value of the turn angle is greater than or equal to 10
+                if abs(self.turns_all[i]) >= 10:
```

Comment:
I think also mininum turn angle should be a parameter that is passed to the detection algorithm.

Reply:
The mininum turn angle (min_turn_angle_deg) is now passed to the detection.